pkg/controller/inferenceservice/resources/knative/service.go

```diff
@@ -111,6 +118,9 @@ func (c *ServiceBuilder) CreatePredictorService(name string, metadata metav1.Obj
 		annotations[autoscaling.MaxScaleAnnotationKey] = fmt.Sprint(predictorSpec.MaxReplicas)
 	}
 
+	if _, ok := annotations[serving.QueueSideCarResourcePercentageAnnotation]; !ok {
```

Comment:
Can we take all of these defaults and assign them in a loop?

Reply:
Take them in a loop won't make it simpler since we still need to keep a map for annotation key and value somewhere. But I do noticed that predictor, transformer and explainer share the same annotations so I moved them to a function to kill some boiler plate.